src/modules/keyboardmanager/KeyboardManagerEditorLibrary/ShortcutControl.cpp

```diff
@@ -352,6 +547,32 @@ void ShortcutControl::AddNewShortcutControlRow(StackPanel& parent, std::vector<s
     }
 }
 
+std::wstring ShortcutControl::RemoveExtraQuotes(const std::wstring& str)
+{
+    if (!str.empty() && str.front() == L'"' && str.back() == L'"')
+    {
+        return str.substr(1, str.size() - 2);
+    }
+
+    return str;
+}
+
+ShortcutControl::ShortcutType ShortcutControl::GetShortcutType(const winrt::Windows::UI::Xaml::Controls::ComboBox& typeCombo)
+{
+    if (typeCombo.SelectedIndex() == 0)
```

Comment:
now that we have `ShortcutType`, perhaps we could assign an `int` value to each variant so we can check for it in places like this for easier reading? 🙂

Reply:
@yuyoyuppe, quick replies (I'm working...)

**_I assume it was done like that to add chords support?_** 
  Actually, I had some knowledge/experience in centralized_kb_hook because of another PR I did that would allow RT-Run to be activated with just `Win` and also NOT break any existing `Win+?` keys. Which was tricky. 

But yes, it worked out for the chords which I coded/test before doing any UI work. 

**_Perhaps for the first version, that should be a global setting in KBM? Though I'm not sure how useful people would find it. @crutkas thoughts?_** 
  It so tricky since some programs like msedge.exe/explorer.exe would always start a new one (with the URL to open as a argument). Others never would want two copies (photoshop/keepass/etc.). You can see some hard coded choices in the PR for now. 

**_I think by default we want to run non-elevated..._** 
I agree and need to dig into this code...

I will look though the rest of the comments later when I have time. THANKS. I don't ever code C++ so I know I write a lot bad code.

Reply:
OK, my C++ is not strong enough! 

**Good'ish:**
I updated "centralized_kb_hook" to use run_non_elevated/run_elevated/run_same_elevation, and that works, awesome and THANKS! I have not yet added a way to use each, but they test fine. I had to update one to allow a start-in-dir (working dir) but other than that I just need to `#include <common/utils/elevation.h>` to `centralized_kb_hook.cpp`. This is only good if we need to use CentralizedKeyboardHook for this, see below.

**Bad'ish 1** 
With regard to using KeyboardEventHandlers instead of CentralizedKeyboardHook, I ran into an issue where KeyboardEventHandlers will not compile if I add `#include <common/utils/elevation.h>`. If it worked we'd be pretty good I think, until we want chords. 

**Bad'ish 2** 
With regard to the  config file watcher, I tried to update KeyboardManager to allow access to the shortcuts but ran into a compile issue when adding `#include "modules/keyboardmanager/KeyboardManagerEngineLibrary/KeyboardManager.h"`

So, I really think we need to ignore the config file watcher for now, and decide if we really need chords and if we can do them in KeyboardEventHandlers or if we really need to do it in CentralizedKeyboardHook. 

I will investigate...

Reply:
In case you're stuck on something, don't hesitate to ping me in this thread 🙂

Reply:
@yuyoyuppe moving to KeyboardEventHandlers  is going nicely, including a new implementation of the chords.  

2 things I think I would like help with. 
1. The ShortcutControl UI, we'll need a select list or something to pick the elevation mode and maybe a file browser to select what program to run. 
2. I would like access to `run_elevated` and `run_non_elevated` in `KeyboardEventHandlers.cpp`. Know how to do that? Including `<common/utils/elevation.h>` breaks the build.

Reply:
> The ShortcutControl UI, we'll need a select list or something to pick the elevation mode and maybe a file browser to select what program to run.

I think we could add a similar `ComboBox` as was done for shortcut type. As for the file browser,  we have a similar functionality on VCM page for picking camera overlay image: `SelectOverlayImage` -> `PickFileDialog`.

> I would like access to run_elevated and run_non_elevated in KeyboardEventHandlers.cpp. Know how to do that? Including <common/utils/elevation.h> breaks the build

Pushed a fix. 🙂

Reply:
I added the "ComboBox as was done for shortcut type", testing that now. I will check the "PickFileDialog", I got "in trouble" when I used the wrong one when did the PT Backup & restore code 😵‍💫

I made local copies of run_elevated and run_non_elevated to test, and it looks like I can clean those up when I get your fix! Sweet.

Reply:
> I updated "centralized_kb_hook" to use run_non_elevated/run_elevated/run_same_elevation

This is not what we need here. The ui drop-down and the code should support "run elevated", "run non elevated" and "ask for credentials"¹. Sure should the first and second use the existing elevation ("run same elevation") if possible.

¹) With ask for credentials/run as (other) user I mean the following:

Reply:
FYI, I am killing all my changes to "centralized_kb_hook" so we can ignore that part.

Right now I only have run_non_elevated/run_elevated in use in KeyboardEventHandlers. 

![image](https://github.com/microsoft/PowerToys/assets/4396667/7f7660b4-e886-4974-bf45-41d8a383dd11)

1. Do we need also need "run as" _different user_ ?
2. Also, how we start the programs might change a little if we can support @htcfreek here https://github.com/microsoft/PowerToys/pull/30121#issuecomment-1843677655